connect/api/src/main/java/org/apache/kafka/connect/data/ConnectSchema.java

```diff
@@ -251,19 +251,30 @@ public static void validateValue(String name, Schema schema, Object value) {
                 break;
             case ARRAY:
                 List<?> array = (List<?>) value;
-                for (Object entry : array)
-                    validateValue(schema.valueSchema(), entry);
+                Schema arrayValueSchema = assertSchemaNotNull(name, "elements", schema.valueSchema());
+                for (Object entry : array) {
+                    validateValue("entry", arrayValueSchema, entry);
```

Comment:
I agree that the field name sucks here. I don't think that justifies keeping the old behavior, especially where `field "null"` appears in the exception :)

I've implemented your suggestion with a new validateValue method, so that existing external callers passing a field name will keep the existing quoting.